src/main/drivers/io_preinit.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "drivers/io.h"
+#include "drivers/resource.h"
+#include "drivers/system.h"
+
+static unsigned preinit_index = 0;
+
+void ioPreinitByIO(const IO_t io, uint8_t iocfg, ioPreinitPinState_e init)
+{
+    if (!io) {
+        return;
+    }
+    preinit_index++;
+
+    IOInit(io, OWNER_PREINIT, preinit_index);
+    IOConfigGPIO(io, iocfg);
+
+    if (init != PREINIT_PIN_STATE_DEFAULT) {
+        IOWrite(io, init == PREINIT_PIN_STATE_LOW);
```

Comment:
PREINIT_PIN_STATE_LOW will set pin high (1)

And PREINIT_PIN_STATE_DEFAULT will keep last value written to  ODR register, it isn't too useful. at least name it PREINIT_PIN_STATE_NOCHANGE or similar ; and it may be alliased as PREINIT_PIN_STATE_NA (or DNC) for input pins

Reply:
I tossed up on no change. 

I also tossed up on not having it, and detecting input mode from iocfg- however that meant GPIO_MODE_INPUT from stm libs bleed out... as iocfg is different packing depending on platform.

No change works for me.